src/core/server/execution_context/execution_context_container.ts

```diff
@@ -50,18 +50,23 @@ export interface IExecutionContextContainer {
 }
 
 function stringify(ctx: KibanaExecutionContext): string {
-  const stringifiedCtx = `${ctx.type}:${ctx.name}:${encodeURIComponent(ctx.id!)}`;
+  const stringifiedCtx = `${encodeURIComponent(ctx.type)}:${encodeURIComponent(
```

Comment:
It's not strictly necessary to encode `type` and `name` since Kibana controls their value. We can even add a runtime validation. OTOH, it doesn't hurt and adds some extra safety.

Reply:
> since Kibana controls their value

Can't any arbitrary consumer just call `core.executionContext.withContext({...someContext, name: '☺☺☺'}, () => { ... })` ?

Reply:
You can. That's why I mentioned a runtime validation. But as said, let's keep it as additional protection.